client/client_test.go

```diff
@@ -98,7 +98,7 @@ func (cs *clientSuite) TestClientSysInfo(c *check.C) {
 func (cs *clientSuite) TestClientReportsOpError(c *check.C) {
 	cs.rsp = `{"type": "error", "status": "potatoes"}`
 	_, err := cs.cli.SysInfo()
-	c.Check(err, check.ErrorMatches, `failed with "potatoes"`)
+	c.Check(err, check.ErrorMatches, `cannot get system information: failed to unmarshal response: unexpected end of JSON input`)
```

Comment:
yeah, that's a problem; it tells the user absolutely nothing about what went wrong :-(

Sometimes the error result will be {}. In those cases you're better off showing the status message. So if the json loading fails, return an errorResult with Str manually set to the status.

Ugly as heck, but it'll let us move forwards on this.